src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -127,9 +260,16 @@ public static object GetObjectForIUnknown(IntPtr pUnk)
         }
 
         [SupportedOSPlatform("windows")]
-        public static object? GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
+        public static unsafe object? GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
         {
-            throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+            if (pSrcNativeVariant == IntPtr.Zero)
+            {
+                return null;
+            }
+
+            Variant* data = (Variant*)pSrcNativeVariant;
+
+            return data->ToObject();
```

Comment:
I think that this has still the problem of hitting infinite recursion. For example, `ToObject` will call `GetObjectForNativeVariant` for `VT_RECORD` variant.

Reply:
I am ok with not supporting the less common variant types like ``VT_RECORD`, but it needs to throw exception instead of infinite recursion.